tripy/tripy/frontend/trace/ops/shape.py

```diff
@@ -63,4 +63,12 @@ def shape(self: "tripy.Tensor") -> List["tripy.DimensionSize"]:
         assert shape == [8, 2]
     """
 
-    return [GetDimensionSize.build([self], dim=index, always_cast_to_dimension_size=True) for index in range(self.rank)]
+    from tripy.frontend.dimension_size import DimensionSize
+
+    # If the shape is statically known, we do not need to insert any operator calls and we can memoize.
+    if self.shape_memo is None and all(dim >= 0 for dim in self.trace_tensor.shape):
+        self.shape_memo = [DimensionSize(dim) for dim in self.trace_tensor.shape]
```

Comment:
I'm not sure there's actually a benefit to storing `shape_memo`. Can't we simplify this to:
```py
if all(dim >= 0 for dim in self.trace_tensor.shape):
    return list(self.trace_tensor.shape)
```
?
I don't think we even need to return `DimensionSize` in this case.

Reply:
It's the signature of the shape operator.

Reply:
Sure, but we could always change the signature. Ideally, `DimensionSize` and `int` behave basically the same from the user's perspective. I'm assuming you have `shape_memo` here because it's expensive to construct `DimensionSize`s?

Reply:
Oh that's right, most of the interfaces will convert `ShapeLike`s anyway. I didn't want to create new `DimensionSize`s constantly, yes. I'll keep it as a tuple of ints if it won't break anything.

Reply:
One place where returning straight ints does break things: In `slice`, some of the preprocessing assumes that the elements of a shape will be tensors. That will just take some wrapping in those cases.

Reply:
We can compare the changes in commit [8e3a4f4b](https://github.com/NVIDIA/TensorRT-Incubator/pull/379/commits/8e3a4f4b7ac71505501ba8964f280bef960975fc). It might be less of a headache to do the conversion in the shape operator than to figure out whether to convert afterwards.

Reply:
That's quite strange - why aren't we able to use integers in these places? That seems like an MLIR-TRT bug.

Reply:
Okay, looks like these can be corrected by handling scalars in a couple of other places. Probably less complicated to do it that way than to wrap `DimensionSize` manually, so I'll change it over.

Reply:
Are we returning floats instead of ints for the shape? Seems like it's turning into a float at some point for whatever reason.